compiler/src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -863,7 +863,7 @@ object Parsers {
           makeTupleOrParens(inParens(argTypes(namedOK = false, wildOK = true)))
         }
       else if (in.token == LBRACE)
-        atPos(in.offset) { RefinedTypeTree(EmptyTree, refinement()) }
+        atPos(in.offset) { inBraces(emptyRefinementOrSingletonExpr()) }
```

Comment:
If you're changing the grammar, it'd be nice to update the grammar comment above (and reflect the update in https://github.com/lampepfl/dotty/blob/master/docs/docs/internals/syntax.md)